site/src/components/pages/hub/BlockDataContainer.tsx

```diff
@@ -58,119 +61,178 @@ export const BlockDataContainer: VoidFunctionComponent<
+              justifyContent: "center",
+              marginRight: 0,
+            },
+            ".MuiTabs-indicator": {
+              backgroundColor: theme.palette.common.white,
+              borderRadius: "6px",
+              position: "absolute",
+              top: 4,
+              bottom: 4,
+              marginLeft: "4px",
+              height: "unset",
+              boxShadow: 1,
+              width: "48% !important",
+            },
+          }}
+        >
+          <Tab label="Preview" />
+          <Tab label="Source Code" />
+        </Tabs>
+      )}
+
+      <Box
+        mt={5}
+        sx={{
+          display: { xs: "flex", md: "grid" },
+          flexDirection: { xs: "column", md: "unset" },
+          gridTemplateColumns: { md: "60% 40%" },
+        }}
+      >
+        <Box
+          sx={{
+            ...(isMobile && {
+              display: activeMobileTab === 0 ? "block" : "none",
+            }),
+          }}
+        >
+          <Box sx={{ height: 320, backgroundColor: "white" }}>
+            <Tabs
+              TabIndicatorProps={{
+                style: { display: "none" },
+              }}
+              sx={{
+                "& .MuiTab-root": {
+                  textTransform: "none",
+                  color: "#64778C",
+                },
+                "& .MuiTab-root.Mui-selected": {
+                  backgroundColor: theme.palette.gray[10],
+                  color: theme.palette.purple[700],
+                  borderTopLeftRadius: 6,
+                  borderTopRightRadius: 6,
+                },
+              }}
+              value={blockTab}
+              onChange={(_event, newValue: number) => setBlockTab(newValue)}
             >
-              {blockModule && <blockModule.default {...props} />}
-            </Box>
-          </TabPanel>
+              <Tab label={metadata.displayName} />
+            </Tabs>
+            <TabPanel value={blockTab} index={0}>
+              <Box
+                display="flex"
+                alignItems="center"
+                sx={{
+                  backgroundColor: "#F7FAFC",
+                  height: "100%",
+                  padding: theme.spacing(0, 4),
+                  overflow: "auto",
+                }}
+              >
+                {blockModule && <blockModule.default {...props} />}
+              </Box>
+            </TabPanel>
+          </Box>
         </Box>
-      </Box>
-      <Box>
-        <BlockDataTabs
-          blockDataTab={blockDataTab}
-          setBlockDataTab={setBlockDataTab}
-        />
-
         <Box
           sx={{
-            position: "relative",
-            borderBottomLeftRadius: 6,
-            borderBottomRightRadius: 6,
+            ...(isMobile && {
+              // @todo use an enum value instead
+              display: activeMobileTab === 1 ? "block" : "none",
+            }),
           }}
         >
-          <BlockDataTabPanels
+          <BlockDataTabs
```

Comment:
In the designs this is a dropdown on mobile, not a tab navigator. If you choose not to add this in this PR please create a follow up task so that it's captured.

Reply:
True. I planned to implement that in a follow-up task.

Reply:
https://app.asana.com/0/1201519755227635/1201664899405923

Reply:
[Task](https://app.asana.com/0/1201519755227635/1201664899405923)